modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataOutputDelegate.java

```diff
@@ -144,4 +149,34 @@
 
     return result.isSuccess() ? success(descriptor) : failure(descriptor, result.getFailures());
   }
+
+  private MetadataType adaptToListIfNecessary(MetadataType resolvedType, Object key, MetadataContext metadataContext)
+      throws MetadataResolvingException {
+    if (!(component.getOutput().getType() instanceof ArrayType)) {
+      return resolvedType;
+    }
+
+    MetadataType outputType = ((ArrayType) component.getOutput().getType()).getType();
+    String typeId = getTypeId(outputType).orElse(null);
```

Comment:
`org.mule.runtime.extension.api.util.ExtensionMetadataTypeUtils#getId` already handles the optional and fallbacks to the classname if missing but of JAVA format (safely)